src/utils/chart.ts

```diff
@@ -125,3 +127,73 @@ export const getFeeRateSamples = (feeRates: Array<FeeRateTracker.TransactionFeeR
 
   return samples
 }
+
+type AssertIsArray = (
+  value: EChartOption.Tooltip.Format | EChartOption.Tooltip.Format[],
+) => asserts value is EChartOption.Tooltip.Format[]
+
+export const assertIsArray: AssertIsArray = (value: EChartOption.Tooltip.Format | EChartOption.Tooltip.Format[]) => {
+  const isArray = Array.isArray(value)
+  if (!isArray) {
+    throw new Error('value is not an array')
+  }
+}
+
+type AssertIsNotArray = (
+  value: EChartOption.Tooltip.Format | EChartOption.Tooltip.Format[],
+) => asserts value is EChartOption.Tooltip.Format
+
+export const assertNotArray: AssertIsNotArray = (
+  value: EChartOption.Tooltip.Format | EChartOption.Tooltip.Format[],
+) => {
+  const isArray = Array.isArray(value)
+  if (isArray) {
+    throw new Error('value is an array')
+  }
+}
+
+type AssertSerialsItem = (value: EChartOption.Tooltip.Format) => asserts value is SeriesItem
+export const assertSerialsItem: AssertSerialsItem = (value: EChartOption.Tooltip.Format) => {
+  if (
+    typeof value.seriesName !== 'string' ||
+    typeof value.name !== 'string' ||
+    typeof value.color !== 'string' ||
+    typeof value.dataIndex !== 'number'
+  ) {
+    throw new Error('invalid SeriesItem')
+  }
+}
+
+export const assertSerialsDataIsString: (value: EChartOption.Tooltip.Format) => asserts value is { data: string } = (
+  value: EChartOption.Tooltip.Format,
+) => {
+  if (typeof value.data !== 'string') {
+    throw new Error('data is not string')
+  }
+}
+
+export const assertSerialsDataIsStringArrayOf3: (
+  value: EChartOption.Tooltip.Format,
+) => asserts value is { data: [string, string, string] } = (value: EChartOption.Tooltip.Format) => {
+  if (
+    Array.isArray(value.data) ||
+    typeof value.data[0] !== 'string' ||
+    typeof value.data[1] !== 'string' ||
+    typeof value.data[2] !== 'string'
```

Comment:
`Array.prototype.every` could be used here

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every